src/GenericForm.js

```diff
@@ -1,80 +1,144 @@
-      childrenObj[el.name]= obj;
+    this.props.fields.forEach(function(el, i) {
+      let obj = { ...el,
+        key: el.name
+      };
+      childrenObj[el.name] = obj;
     });
-    this.state ={
+    this.state = {
       isValid: false,
       isComplete: false,
       fields: childrenObj
     };
   }
   handleFieldChange(field, value) {
-    console.log("We are updating '"+ field + "' with: '"+ value);
+    console.log("We are updating '" + field + "' with: '" + value);
     let updatedFields = this.state.fields;
     updatedFields[field].value = value;
-    this.setState({ fields: updatedFields });
+    this.setState({
+      fields: updatedFields
+    });
   }
   render() {
-    let childNodes = [];
-    let fields = this.state.fields;
+      let childNodes = [];
+      let fields = this.state.fields;
+
+      let childButtons = [];
+      let buttons = this.props.buttons;
 
-    for (var key in fields) {
-      if (fields.hasOwnProperty(key)) {
-        let el = fields[key];
-        switch (el.type) {
-          case "text":
-            childNodes.push(<CustomInput {...el} onChange={this.handleFieldChange} />)
-            break;
-          case "password":
-            childNodes.push(<CustomPassword {...el} onChange={this.handleFieldChange} />)
-            break;
-          case "textarea":
-            childNodes.push(<CustomTextarea {...el} onChange={this.handleFieldChange} />)
-            break;
-          case "radio":
-            childNodes.push(<CustomRadio {...el} onChange={this.handleFieldChange} />)
-            break;
-          case "checkbox":
-            childNodes.push(<CustomCheckbox {...el} onChange={this.handleFieldChange} />)
-            break;
-          case "select":
-            childNodes.push(<CustomSelect {...el} onChange={this.handleFieldChange} />)
-            break;
-          default:
-            childNodes.push(<CustomInput {...el} onChange={this.handleFieldChange} />)
+      for (let id in buttons) {
+        if (buttons.hasOwnProperty(id)) {
+          let el = buttons[id];
+          childButtons.push( < CustomButton { ...el
+            }
+            />)
+          }
         }
-      }
-    }
 
-    return (
-      <div className="form-container">
-        <form onSubmit={this.onSubmit}>
-          <h2 className="form-title">{this.props.title}</h2>
-          {childNodes}
-          <button className="submit-button">Submit</button>
-        </form>
-      </div>
-    );
-  }
-  onSubmit() {
-      console.log(this.state.fields);
-      this.props.onSubmit(this.state.fields);
-  }
-}
+        for (var key in fields) {
+          if (fields.hasOwnProperty(key)) {
+            let el = fields[key];
+            switch (el.type) {
+              case "text":
+                childNodes.push( < CustomInput { ...el
+                    }
+                    onChange = {
+                      this.handleFieldChange
+                    }
+                    />)
+                    break;
```

Comment:
there is a problem with the indentation here

Reply:
done